### PR TITLE
[10.0] [MIG] Migration of server_environment_ir_config_parameter

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by sbidoul** [comment]
@rousseldenis can you please forward port this too https://github.com/OCA/server-tools/pull/676 ?

File: server_environment_ir_config_parameter/models/ir_config_parameter.py [hunk 1]
```diff
@@ -51,4 +51,4 @@ def write(self, vals):
                 newvals = dict(vals, value=serv_config.get(SECTION, key))
             else:
                 newvals = vals
-            super(IrConfigParameter, rec).write(newvals)
+        return super(IrConfigParameter, rec).write(newvals)
```
Comment (sbidoul) [inline server_environment_ir_config_parameter/models/ir_config_parameter.py]: @rousseldenis this does not look right. The write must be inside the loop as we overwrite whatever is provided in vals with the values found in the config file, so each property may end up with a different value. So this commit changes the expected behaviour.

File: server_environment_ir_config_parameter/models/ir_config_parameter.py [hunk 2]
```diff
@@ -51,4 +51,5 @@ def write(self, vals):
                 newvals = dict(vals, value=serv_config.get(SECTION, key))
             else:
                 newvals = vals
-        return super(IrConfigParameter, rec).write(newvals)
+            super(IrConfigParameter, rec).write(newvals)
+        return True
```
Comment (sbidoul) [inline server_environment_ir_config_parameter/models/ir_config_parameter.py]: Ah yes, fixed in next commit, okay.

**Comment by lmignon** [comment]
This one depends of #595

**Comment by lmignon** [comment]
@sbidoul #676 forward ported

**Review by lmignon — APPROVED** [review]
Code review + test

**Comment by sbidoul** [comment]
rebased

**Comment by sbidoul** [comment]
This one is now finished and migrated to new API.

**Review by simahawk — APPROVED** [review]

**Review by leemannd — APPROVED** [review]
Code review and tested. Thank you! :+1:

**Review by atchuthan — APPROVED** [review]
code review :+1:

**Review by guewen — APPROVED** [review]
Nice!

**Pull request merged**